### PR TITLE
[Feature] VAR Inference and Training

### PR DESCRIPTION
**Pull request by Songyuanwei**
# What does this PR do?

Fixes # (issue)

Adds # (feature)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [ ] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

File: examples/var/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+
+## Requirements
+
+| mindspore | ascend driver | firmware    | cann toolkit/kernel |
+|:---------:|:-------------:|:-----------:|:-------------------:|
+| 2.5.0     | 24.1.RC3    | 7.3.0.1.231 | 8.0.RC3.beta1       |
+
+```shell
+pip install -r requirements.txt
```
Comment (CaitinZhao) [inline examples/var/README.md:22]:
```suggestion
cd mindone
pip install -e .[training]
cd examples/emu3
pip install -r requirements.txt
```
and delete the sys.path.insert in python file

File: examples/var/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Pillow
+numpy==1.24
```
Comment (CaitinZhao) [inline examples/var/requirements.txt:2]: why numpy must be 1.24?

File: examples/var/README.md (hunk 1 above)
Comment (Songyuanwei) [reply examples/var/README.md:25]: done

File: examples/var/requirements.txt (hunk 2 above)
Comment (Songyuanwei) [reply examples/var/requirements.txt]: No numpy version required.  fixed.

**Review by CaitinZhao — APPROVED** [review]

File: examples/var/tools/ms_vae.txt
Comment (SamitHuang) [inline examples/var/tools/ms_vae.txt:1]: if use mint, there is no need to re-map the parameter name.

File: examples/var/tools/convert_weight.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    with open(ms_param, "r") as f:
+        f_ms_param = f.readlines()
+    pt_weight = torch.load(pt_weight, map_location=torch.device("cpu"))
+    if "state_dict" in pt_weight:
+        pt_weight = pt_weight["state_dict"]
+
+    ms_weight = []
+
+    assert len(f_pt_param) == len(f_ms_param)
+
```
Comment (SamitHuang) [inline examples/var/tools/convert_weight.py:30]: there is no need to do name mapping. can remove the additional 4 txt files.
```
ms_weight = []
for pname in pt_weight:
    ms_name = pname
    ms_val = pt_weight[pt_name].cpu().detach().numpy()
    ms_weight.append({"name": ms_name, "data": ms.Tensor(ms_val)})
```

Comment (Songyuanwei) [reply examples/var/tools/convert_weight.py]: done

File: examples/var/tools/ms_vae.txt
Comment (Songyuanwei) [reply examples/var/tools/ms_vae.txt:1]: done

**Review by SamitHuang — APPROVED** [review]
